### PR TITLE
[FIL-247] Containers Overflow

### PR DESCRIPTION
**Pull request by artyom-yurin**
### Description of the Change
Made Markets safe for big data

### Benefits
Fuhon can store and work with bigger sectors

### Possible Drawbacks 
Many temp files

### CODE REVIEW

File: core/storage/car/car.cpp [hunk 1]
```diff
@@ -34,6 +40,19 @@ namespace fc::storage::car {
     return std::move(header.roots);
   }
 
+  outcome::result<std::vector<CID>> loadCar(Ipld &store,
+                                            const std::string &car_path) {
+    mapped_file car_file(car_path);
+    if (!car_file.is_open()) {
+      return outcome::success();
```
Comment (turuslan) [inline core/storage/car/car.cpp:47]: Did you mean error?

File: core/storage/car/car.cpp [hunk 2]
```diff
@@ -95,4 +114,72 @@ namespace fc::storage::car {
...
   }
+
+  void writeUvarint(std::ostream &output, uint64_t value) {
+    output << common::span::bytestr(libp2p::multi::UVarint{value}.toBytes());
+  }
+
+  void writeHeader(std::ostream &output, const std::vector<CID> &roots) {
+    OUTCOME_EXCEPT(bytes, codec::cbor::encode(CarHeader{roots, CarHeader::V1}));
+    writeUvarint(output, bytes.size());
+    std::copy(bytes.begin(),
```
Comment (turuslan) [inline core/storage/car/car.cpp:125]: Why not `output << common::span::bytestr(bytes)`?

File: core/markets/retrieval/provider/impl/retrieval_provider_impl.hpp [hunk 3]
```diff
@@ -32,8 +32,11 @@ namespace fc::markets::retrieval::provider {
   using primitives::SectorNumber;
   using primitives::piece::PieceData;
   using primitives::piece::UnpaddedPieceSize;
+  using storage::Path;
   using GsResStatus = ::fc::storage::ipfs::graphsync::ResponseStatusCode;
 
+  const Path kFilestoreTempDir = "/tmp/fuhon/retrieval-market/";
```
Comment (Alexey-N-Chernyshov) [inline core/markets/retrieval/provider/impl/retrieval_provider_impl.hpp:38]: May be move to config?

File: test/core/storage/car/car_test.cpp [hunk 4]
```diff
@@ -37,6 +39,15 @@ TEST(CarTest, LoadSuccess) {
           "0171a0e402202d4c00840d4227f198ec6c5343b8c70af7f008a7775d67393d10430ea2fa012f"_cid));
 }
 
+TEST(CarTest, LoadFromFileSuccess) {
```
Comment (Alexey-N-Chernyshov) [inline test/core/storage/car/car_test.cpp:42]: Please, add test comments.

File: test/core/storage/car/car_test.cpp [hunk 5]
```diff
@@ -104,3 +115,24 @@ TEST(SelectiveCar, MakeSelectiveCar) {
                                          << "expected" << std::endl
                                          << expected_car << std::endl;
 }
+
+TEST(SelectiveCar, MakeSelectiveCarToFile) {
```
Comment (Alexey-N-Chernyshov) [inline test/core/storage/car/car_test.cpp:119]: Please, add test comments.

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: test/core/storage/car/car_test.cpp (hunk 5 above)
Comment (artyom-yurin) [reply test/core/storage/car/car_test.cpp:131]: Added

File: test/core/storage/car/car_test.cpp (hunk 4 above)
Comment (artyom-yurin) [reply test/core/storage/car/car_test.cpp:47]: Added

File: core/markets/retrieval/provider/impl/retrieval_provider_impl.hpp (hunk 3 above)
Comment (artyom-yurin) [reply core/markets/retrieval/provider/impl/retrieval_provider_impl.hpp:38]: Moved

File: core/storage/car/car.cpp (hunk 2 above)
Comment (artyom-yurin) [reply core/storage/car/car.cpp]: Fixed

File: core/storage/car/car.cpp (hunk 1 above)
Comment (artyom-yurin) [reply core/storage/car/car.cpp]: Yes, thank you

**Review by turuslan — APPROVED** [review]

**Pull request merged**